src/main/java/org/jboss/xavier/integrations/route/RouteBuilderExceptionHandler.java

```diff
@@ -0,0 +1,36 @@
+package org.jboss.xavier.integrations.route;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.jboss.xavier.integrations.jpa.service.AnalysisService;
+
+import javax.inject.Inject;
+import java.util.Map;
+
+import static org.jboss.xavier.integrations.route.MainRouteBuilder.ANALYSIS_ID;
+import static org.jboss.xavier.integrations.route.MainRouteBuilder.MA_METADATA;
+
+public abstract class RouteBuilderExceptionHandler extends RouteBuilder {
+    @Inject
+    protected AnalysisService analysisService;
+
+    @Override
+    public void configure() throws Exception {
+        onException(Exception.class).routeId("exception-handler")
+                .handled(true)
+                .process(this::markAnalysisAsFailed)
+                .stop();
+    }
+
+    public void markAnalysisAsFailed(Exchange e) {
+        try {
+            String analysisId = e.getIn().getHeader(ANALYSIS_ID, "", String.class);
+            if (analysisId.isEmpty()) {
+                analysisId = (String) e.getIn().getHeader(MA_METADATA, Map.class).get(ANALYSIS_ID);
+            }
+            analysisService.markAsFailedIfNotCreated(Long.parseLong(analysisId));
```

Comment:
It seems like there's no need to check for `analysisId` to not be null which is good if we're sure it has always a value. Is this true?

Reply:
+1